### PR TITLE
goverlay: 0.7.1 -> 1.2

### LINKED ISSUE #335978: Package request: Goverlay 0.7.1 -> 1.2
Update request: Goverlay 0.7.1 -> 1.2

  -  Package name: Goverlay
  - Latest released version: 0.7.1

   - Current version on the unstable channel: 0.7.1
   - Current version on the stable/release channel: 0.7.1
   
- [x]   Checked the nixpkgs pull requests (there is a failed pull request to v0.9.1)

* homepage URL: https://github.com/benjamimgois/goverlay
* source URL: https://github.com/benjamimgois/goverlay/
* license: GPL-3.0
* platforms: linux

 **Notify maintainers**

**@kira-bruneau** 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Chais**
- Updated GOverlay from `0.7.1` to `1.2`
   - Dropped unnecessary patches
   - Added new ones
   - Changelogs for versions 0.8.1 upwards can be viewed [here](https://github.com/benjamimgois/goverlay/releases). Older changelogs seem to be lost to issues with account access.
- Resolves #335978
- Depends on #368201 for `lazarus-qt6` and `libqt6pas`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/graphics/goverlay/default.nix
Comment (SigmaSquadron) [inline pkgs/tools/graphics/goverlay/default.nix:25]: For consistency with other packages, please put `lib` as the first attribute in the function call, followed by `stdenv` and `fetchFromGitHub`.

**Review by SigmaSquadron — COMMENTED** [review]
Welcome to Nixpkgs! Please review the [Contribution Guidelines](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) for more information about how to format your commits. Notably, their titles should follow the `packagename: oldversion -> newversion` template. Ideally, the removal of `find-xdg-data-files` and the alphabetical sorting should also be separate commits, following the `packagename: change description` template.

File: pkgs/tools/graphics/goverlay/default.nix
Comment (Chais) [reply pkgs/tools/graphics/goverlay/default.nix:25]: The sorting I can split out, I'm not sure how useful the removal of `find-xdg-data-files` is on its own. It was required with 0.7.1 but isn't with 1.2. Unless you're referring of the deletion of `find-xdg-data-files.patch`, than I can easily separate.

**Comment by SigmaSquadron** [comment]
Looks good to me! The commits have been named appropriately and the diff looks alright. I'll wait until the other PR is merged before running a build.

**Comment by kira-bruneau** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 368202`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>goverlay</li>
    <li>goverlay.man</li>
  </ul>
</details>

**Comment by kira-bruneau** [comment]
Thanks for the update @Chais!!

Just for reference, I was originally getting a build error:

```An unhandled exception occurred at $0000000000444A56:
EThread: Failed to create new thread
  $0000000000444A56
  $0000000000444E71
  $0000000000937974
  $0000000000935AD8
  $000000000052F9BA
  $00000000005AAB8B
  $00000000004056F6
  $00000000004051E3
  $000000000040343F
  $0000000000408CE3
  $000000000040B9C9
```

This was one of the reasons why [I removed myself as maintainer on goverlay](https://github.com/NixOS/nixpkgs/pull/351523), but finally tracked the issue down to a bug in lazarus: https://bugs.gentoo.org/937421.

I just had `nix.daemonCPUSchedPolicy = "idle"` set in my Nix config. Temporarily removing this allows me to build the package again.

**Review by kira-bruneau — APPROVED** [review]

**Review by kira-bruneau — COMMENTED** [review]
Oh wait, I missed that CI is failing on formatting right now. Other than that though, this looks good to me!

**Comment by Chais** [comment]
> Oh wait, I missed that CI is failing on formatting right now. Other than that though, this looks good to me!

@kira-bruneau I saw the formatting check failing but was unsure if it's something that should be remedied, given that `nixfmt` mostly pulls arrays into fewer lines and most files have one element per line.

**Comment by kira-bruneau** [comment]
> @kira-bruneau I saw the formatting check failing but was unsure if it's something that should be remedied, given that nixfmt mostly pulls arrays into fewer lines and most files have one element per line.

Oh yep it should be remedied here. It's not complete yet, but there's been a big push to have everything formatted with nixfmt. There are still some files that haven't been formatted yet, but that's mainly to avoid merge conflicts.

See https://github.com/NixOS/nixpkgs/pull/322537

**Comment by Chais** [comment]
`nixfmt` check still fails after running it. Would be helpful to know what the check takes offense at.

**Comment by kira-bruneau** [comment]
Oh I just ran it locally and pushed the diff. It looks like the issue was just an extra empty line.

**Comment by Chais** [comment]
> Oh I just ran it locally and pushed the diff. It looks like the issue was just an extra empty line.

Oh, that's interesting. Yours arranged the arrays in one element per line again. What was it originally complaining about then?

**Comment by kira-bruneau** [comment]
Oh it was just the extra blank line above `stdenv.mkDerivation` that was left after removing `find-xdg-data-files`.

**Review by kira-bruneau — APPROVED** [review]

**Pull request merged**
